src/code/z_fireobj.c

```diff
@@ -75,11 +75,11 @@ void FireObj_StepSize(FireObj* fire) {
             FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_2);
         }
     } else if ((fire->state == FIRE_STATE_1) && (Math_StepToF(&fire->dynamicSize, 0.0f, fire->dynamicSizeStep) != 0)) {
-        FireObj_SetState(fire, fire->dynamicSizeStep, 3);
+        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_3);
     }
     if (fire->sizeGrowsCos2 == 1) {
         if ((fire->state == FIRE_STATE_0) || (fire->state == FIRE_STATE_1)) {
-            fire->xScale = (1.0f - Math_CosS((fire->dynamicSize * fire->dynamicSize * 16384.0f))) * fire->size;
+            fire->xScale = (1.0f - Math_CosS(SQ(fire->dynamicSize) * 16384.0f)) * fire->size;
```

Comment:
16384 is 128^2, would it match inside the SQ?

Reply:
So I can't fold it into the `SQ` with the `fire->dynsamicSize`, but could be its own `SQ`. However at that point I wonder if its worth it. `16384.0f` feels like it is more informative than `SQ(128.0f)` especially since it really feels like this is some kind of angle conversion, though can't find one that matches currently.

Reply:
Well, if one thing's squared, I can see that the other would be. It's true that it's pretty nonsensical to square an angle... this is actually the [versine](https://en.wikipedia.org/wiki/Versine), although I don't know why they're using it.

Reply:
Per Discord, I changed my mind on this, would rather use 0x4000, since it's sort of acting like a right angle?